src/hiro_graph_client/clientlib.py

```diff
@@ -699,8 +710,15 @@ def __init__(self,
         :param log_communication_on_error: Log socket communication when an error (status_code of HTTP Response) is
                detected. Default is not to do this.
         :param max_tries: Max tries for BACKOFF. Default is 2.
+        :param pool_maxsize: Size of a connection pool for a single connection. Default is 10.
         """
+        session = requests.Session()
+        adapter = requests.adapters.HTTPAdapter(pool_maxsize=pool_maxsize or self._pool_maxsize)
+        session.mount('https://', adapter)
```

Comment:
afaik mount() is not thread safe in session. is thread-safety guaranteed in here?